packages/binarytree/src/binaryTree.ts

```diff
@@ -0,0 +1,707 @@
+          result.remaining = []
+          return result
+        }
+        // Otherwise, record the unmatched tail of the key.
+        result.remaining = keyInBits.slice(matchingKeyLength)
+        result.stack.push([decodedNode, childNode.path])
+        return result
+      }
+      // Otherwise, push this internal node and continue.
+      result.stack.push([decodedNode, keyInBits.slice(0, matchingKeyLength)])
+      this.DEBUG &&
+        this.debug(
+          `Partial Path ${keyInBits.slice(0, matchingKeyLength)} - found next node in path ${bytesToHex(
+            this.merkelize(decodedNode),
+          )}.`,
+          ['find_path'],
+        )
+
+      // If the decoded node is not internal, then we cannot traverse further.
+      if (!isInternalBinaryNode(decodedNode)) {
+        result.remaining = keyInBits.slice(matchingKeyLength)
+        finished = true
+        break
+      }
+      // The next branch is determined by the next bit after the matched prefix.
+      const childIndex = keyInBits[matchingKeyLength]
+      childNode = decodedNode.getChild(childIndex)
+      if (childNode === null) {
+        result.remaining = keyInBits.slice(matchingKeyLength)
+        finished = true
+      }
+    }
+    this.DEBUG &&
+      this.debug(
+        `Found partial path ${keyInBits.slice(256 - result.remaining.length)} but sought node is not present in trie.`,
+        ['find_path'],
+      )
+    return result
+  }
+
+  /**
+   * Deletes a given `key` from the tree.
+   * @param stem - the stem of the stem node to delete from
+   * @param suffixes - the suffixes to delete
+   * @returns A Promise that resolves once the key is deleted.
+   */
+  async del(stem: Uint8Array, suffixes: number[]): Promise<void> {
+    this.DEBUG && this.debug(`Stem: ${bytesToHex(stem)}; Suffix(es): ${suffixes}`, ['del'])
+    await this.put(stem, suffixes, new Array(suffixes.length).fill(null))
+  }
+
+  /**
+   * Create empty root node for initializing an empty tree.
+   */
+  async createRootNode(): Promise<void> {
+    const rootNode = null
+    this.DEBUG && this.debug(`No root node. Creating new root node`, ['initialize'])
+    this.root(this.merkelize(rootNode))
+    await this.saveStack([[this.root(), rootNode]])
+    return
+  }
+
+  /**
+   * Creates the initial node from an empty tree.
+   * @private
+   */
+  protected async _createInitialNode(
+    stem: Uint8Array,
+    indexes: number[],
+    values: (Uint8Array | null)[],
+  ): Promise<void> {
+    const initialNode = StemBinaryNode.create(stem)
+    for (let i = 0; i < indexes.length; i++) {
+      initialNode.setValue(indexes[i], values[i])
+    }
+    this.root(this.merkelize(initialNode))
+    await this._db.put(this.root(), initialNode.serialize())
+    await this.persistRoot()
+  }
+
+  /**
+   * Saves a stack of nodes to the database.
+   *
+   * @param putStack - an array of tuples of keys (the partial path of the node in the trie) and nodes (BinaryNodes)
+   */
```

Comment:
Shouldn't we be writing zeros here instead of null if we don't allow true deletions at this point?